packages/next/server/lib/squoosh/codecs.ts

```diff
@@ -11,37 +44,47 @@ import mozDec from './mozjpeg/mozjpeg_node_dec.js'
 const mozDecWasm = path.resolve(__dirname, './mozjpeg/mozjpeg_node_dec.wasm')
 
 // WebP
+import type { WebPModule as WebPEncodeModule } from './webp/webp_enc'
 // @ts-ignore
 import webpEnc from './webp/webp_node_enc.js'
 const webpEncWasm = path.resolve(__dirname, './webp/webp_node_enc.wasm')
 // @ts-ignore
 import webpDec from './webp/webp_node_dec.js'
 const webpDecWasm = path.resolve(__dirname, './webp/webp_node_dec.wasm')
 
+// AVIF
+import type { AVIFModule as AVIFEncodeModule } from './avif/avif_enc'
+// @ts-ignore
+import avifEnc from './avif/avif_node_enc.js'
+const avifEncWasm = path.resolve(__dirname, './avif/avif_node_enc.wasm')
+// @ts-ignore
+import avifDec from './avif/avif_node_dec.js'
+const avifDecWasm = path.resolve(__dirname, './avif/avif_node_dec.wasm')
+
 // PNG
 // @ts-ignore
 import * as pngEncDec from './png/squoosh_png.js'
 const pngEncDecWasm = path.resolve(__dirname, './png/squoosh_png_bg.wasm')
-const pngEncDecInit = () =>
-  pngEncDec.default(fsp.readFile(pathify(pngEncDecWasm)))
+const pngEncDecPromise = pngEncDec.default(fsp.readFile(pathify(pngEncDecWasm)))
 
 // OxiPNG
 // @ts-ignore
 import * as oxipng from './png/squoosh_oxipng.js'
 const oxipngWasm = path.resolve(__dirname, './png/squoosh_oxipng_bg.wasm')
-const oxipngInit = () => oxipng.default(fsp.readFile(pathify(oxipngWasm)))
+const oxipngPromise = oxipng.default(fsp.readFile(pathify(oxipngWasm)))
 
 // Resize
 // @ts-ignore
 import * as resize from './resize/squoosh_resize.js'
 const resizeWasm = path.resolve(__dirname, './resize/squoosh_resize_bg.wasm')
-const resizeInit = () => resize.default(fsp.readFile(pathify(resizeWasm)))
+const resizePromise = resize.default(fsp.readFile(pathify(resizeWasm)))
```

Comment:
My concern here is all those WASM modules will always be initialized immediately. Previously with `() => resize.default(fsp.readFile(pathify(resizeWasm)))`, they were only initialized lazily.

For example if the resize processor isn't used, `squoosh_resize_bg.wasm` will **still** be loaded in this worker. Since the workers are keyed by the task it handles: https://github.com/vercel/next.js/blob/92c2dd4971c8ff04ecdc0ad5948a58c95391376c/packages/next/server/lib/squoosh/main.ts#L16-L25

Previously, it ensures that 1 worker only loads and handles 1 type of task, for example: https://github.com/vercel/next.js/blob/92c2dd4971c8ff04ecdc0ad5948a58c95391376c/packages/next/server/lib/squoosh/main.ts#L64

This worker has only loaded the `encodeJpeg` related WASM module inside. And tasks of `encodeJpeg` will always be assigned to this same worker. That ensures that we don't allocate the memory for every WASM module inside every worker.